api/status.go

```diff
@@ -10,20 +10,38 @@
 package api
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
 	"net/http"
 
-	"github.com/labstack/echo"
+	"go.uber.org/zap"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	api "github.com/topfreegames/podium/proto/podium/api/v1"
 )
 
 // StatusHandler is the handler responsible for reporting podium status
-func StatusHandler(app *App) func(c echo.Context) error {
-	return func(c echo.Context) error {
-		c.Set("route", "Status")
-		payload := map[string]interface{}{
-			"app": map[string]interface{}{
-				"errorRate": app.Errors.Rate(),
-			},
-		}
-		return c.JSON(http.StatusOK, payload)
+func (app *App) statusHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "application/json")
+	payload := map[string]interface{}{
```

Comment:
Suggestion: instead of using map[string]interface{}, consider defining a struct that matches your payload and marshaling it instead.

Eg:

```golang
type statusPayload struct {
  app []*statusApp
}
type statusApp struct {
  errorRate double
}
```